### PR TITLE
[Documentation] Concepts: Products

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/book/products.rst [hunk 1]
```diff
@@ -4,69 +4,113 @@
...
+
+Firstly let's learn how to prepare an exemplary Option and its values.
+
+.. code-block::
+
+     /* @var $option OptionInterface */
+     $option = $this->get('sylius.factory.product_option')->createNew();
+     $option->setCode('t_shirt_color');
+
+     $option->setFallbackLocale('en_US');
```
Comment (pjedrzejewski) [inline docs/book/products.rst]: That should not be necessary, hm? See TranslatableFactory that is used for all translatable resources.

File: docs/book/products.rst [hunk 2]
```diff
@@ -4,69 +4,113 @@
...
 
-* *name* - The full name of the product
-* *slug* - Urlized version of the name
-* *description* - Description of the product
-* *shortDescription* - Simple description of the product for lists and banners
-* *metaDescription* - Description for search engines
-* *metaKeywords* - SEO keywords
-* *createdAt* - Date of creation
-* *updateAt* - Timestamp of the most recent update
+.. note::
```
Comment (NeverResponse) [inline docs/book/products.rst]: More of a warning than a note

File: docs/book/products.rst [hunk 3]
```diff
@@ -4,69 +4,113 @@
 Products
 ========
 
-**Product** model represents unique products in your Sylius store. Every product can have different variations or attributes and has following values:
+**Product** model represents unique products in your Sylius store. Every product can have different variations or attributes.
```
Comment (NeverResponse) [inline docs/book/products.rst]: ``and attributes.`` - or suggests you can use only one at a time

File: docs/book/products.rst [hunk 4]
```diff
@@ -4,69 +4,113 @@
...
-* *updateAt* - Timestamp of the most recent update
+.. note::
+
+   Each product has to have at least one variant to be sold in the shop.
 
 Options
 -------
 
-In many cases, you will have product with different variations. The simplest example would be a T-Shirt available in different sizes and colors.
+In many cases, you will want to have product with different variations.
```
Comment (NeverResponse) [inline docs/book/products.rst]: `will` => `would`, `product` => `a product` if i'm not mistaken that you mean singular in this context

File: docs/book/products.rst [hunk 5]
```diff
@@ -4,69 +4,113 @@
...
     * Blue
 
+After defining possible options for a product let's move on to **Variants** which are in fact combinations of options.
+
 Variants
 --------
 
 **ProductVariant** represents a unique combination of product options and can have its own pricing configuration, inventory tracking etc.
 
 You are also able to use product variations system without the options at all.
```
Comment (NeverResponse) [inline docs/book/products.rst]: This line seems misleading. Earlier on it was stated that `Variants` are combinations of options, and yet here it says that you can ignore options altogether.

File: docs/book/products.rst [hunk 6]
```diff
@@ -4,69 +4,113 @@
...
+         foreach ($values as $locale => $value) {
+             $optionValue->setFallbackLocale($locale);
+             $optionValue->setCurrentLocale($locale);
+             $optionValue->setValue($value);
+         }
+
+         $option->addValue($optionValue);
+     }
+
+After you have an Option created and you keep it as ``$option`` variable let's create a Product.
```
Comment (NeverResponse) [inline docs/book/products.rst]: This sentence seems off, but i'm not sure why 😄

File: docs/book/products.rst (hunk 5 above)
Comment (lchrusciel) [inline docs/book/products.rst]: I would even say, that a variant concept should be described before an option part. Variants doesn't have to be based on options, but options are available via variants.

File: docs/book/products.rst [hunk 7]
```diff
@@ -4,69 +4,113 @@
...
 **ProductVariant** represents a unique combination of product options and can have its own pricing configuration, inventory tracking etc.
 
 You are also able to use product variations system without the options at all.
 
-Master Variant
-``````````````
+How to create a Product with options and variants?
+--------------------------------------------------
+
+Firstly let's learn how to prepare an exemplary Option and its values.
```
Comment (lchrusciel) [inline docs/book/products.rst]: I think we should present creating a new product without anything at first section and then extend it with next chapters.

File: docs/book/products.rst [hunk 8]
```diff
@@ -4,69 +4,113 @@
...
+
+After you have an Option created and you keep it as ``$option`` variable let's create a Product.
 
-Each product has a master variant, which tracks the same information as any other variant. It exists to simplify the internal Sylius logic. Whenever a product is created, a master variant for that product will be created too.
+.. code-block:: php
 
-Attributes
-----------
+     /** @var ProductFactoryInterface $productFactory **/
+     $productFactory= $this->get('sylius.factory.product');
```
Comment (lchrusciel) [inline docs/book/products.rst]: Missing space before `=`

**Comment by lchrusciel** [comment]
Just to sum up. I would say, that we should create a basic product and a simple product(with one variant in a firs section and then extend a basic one with additional features. Creating an option should be extracted to some other "Concept" chapter and here we can only link to it. Especially, that it would be worth to mention a difference between an option and option value and so on.

**Comment by pjedrzejewski** [comment]
Thank you Magda! 👍

File: docs/book/products.rst [hunk 9]
```diff
@@ -4,69 +4,161 @@
...
+After being added via the repository, your product will be in the system. But the customers won't be able to buy it.
+
+Variants
+========
+
+**ProductVariant** represents a unique kind of product and can have its own pricing configuration, inventory tracking etc.
+
+Variants may be created out of Options of the product, but you are also able to use product variations system without the options at all.
+
+How to create a Product with a Variant?
```
Comment (lchrusciel) [inline docs/book/products.rst]: It can be done directly from the ProductFactory if you just call a `createWithVariant` function.

**Pull request merged**